### PR TITLE
Implementing job that allow to clean old executions

### PR DESCRIPTION
**Pull request by carlosrfranco**
This job will be executed according to what is configured in the project settings.

### CODE REVIEW

**Comment by carlosrfranco** [comment]
1 - List of executions:
selecao_007

2 - Project settings to enable history cleaner:
selecao_003

3 - History cleaner settings:
 property 1 -> Maximum of days to keep the executions
 property 2 -> When the cleaner will be executed. Is needed to make sure it will not generate a performance issue
selecao_004

4 - Project config file:
selecao_010

5 - List of the executions cleaned:
selecao_011

**Comment by carlosrfranco** [comment]
Log output when cleaner executions history is executed:

selecao_012

selecao_014

**Comment by miguelantonio** [comment]
final review

**Comment by miguelantonio** [comment]
Hi @gschueler just following up on this merge, is there anything you'll like to change before merging?

**Comment by carlosrfranco** [comment]
Is needed to set a feature `rundeck.feature.cleanExecutionsHistoryJob.enabled = true`

File: rundeckapp/grails-app/conf/application.groovy [hunk 1]
```diff
@@ -36,6 +36,7 @@ environments {
         application.refreshDelay=5000
         grails.profiler.disable=false
         feature.incubator.'*'=true
+        rundeck.feature.cleanExecutionsHistoryJob.enabled = true
```
Comment (jtobard) [inline rundeckapp/grails-app/conf/application.groovy]: This feature should be disabled by default by now.

File: rundeckapp/grails-app/init/rundeckapp/BootStrap.groovy [hunk 2]
```diff
@@ -324,6 +324,13 @@ class BootStrap {
          if(!maxLastLines || !(maxLastLines instanceof Integer) || maxLastLines < 1){
              grailsApplication.config.rundeck.gui.execution.tail.lines.max = 500
          }
+         if(grailsApplication.config.rundeck.feature.cleanExecutionsHistoryJob.enabled){
+             println("cleanExecutionsHistoryJob ${grailsApplication.config.rundeck.feature.cleanExecutionsHistoryJob.enabled}")
```
Comment (jtobard) [inline rundeckapp/grails-app/init/rundeckapp/BootStrap.groovy]: this `println` should be a `log.debug` or `log.info`

**Comment by ahonor** [comment]
* All fields should be required so no blank fields accepted.
* Default days to keep executions: 60
*  Minimum executions to keep: 50

image

**Comment by lukasged** [comment]
thanks for this feature 👍 

I was testing it in Rundeck 3.1 and it works very fast: only takes some seconds to delete 1K jobs and without requiring too much CPU.

But when I check the folder with the logs, I see that the `*.rdlog` and `*.state.json` files are cleaned up correctly, but the `*.execution.xml` are all still there (I had around 3K jobs and after cleanup, there are 3K of these xml files). 

Is this the expected behaviour?

**Pull request merged**